### ISSUE TITLE
[wasm] Regression. mono-aot-cross asserts with function:mono_method_to_ir, VAR 1 (TValue_REF) cannot be expanded in this context with 1 instantiations

### ISSUE THREAD

**Issue by Xarlot** [dotnet/runtime#59739]
### Description
The issue [[wasm] mono-aot-cross asserts with function:mono_method_to_ir, VAR 1 (TValue_REF) cannot be expanded in this context with 1 instantiations](https://github.com/dotnet/runtime/issues/57560#) was marked as fixed in net 6 RC 1 but still can be reproduced with the project from the issue. https://github.com/dotnet/runtime/files/6998976/wasmtest.zip

Additionally, I have a problem with our project that looks pretty the same. 
```

 Mono: Request to load System.Text.RegularExpressions in alc 000001F6C785FFA0
  Mono: Assembly already loaded in the active ALC: 'System.Text.RegularExpressions'.
  Mono: Assembly Ref addref netstandard[000001F6C78949D0] -> System.Text.RegularExpressions[000001F6C7894570]: 3
  * Assertion at D:\workspace\_work\1\s\src\mono\mono\mini\method-to-ir.c:7329, condition `is_ok (error)' not met, function:mono_method_to_ir, VAR 1 (TKey_REF) cannot be expanded in this context with 1 instantiations
C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Sdk\6.0.0-rc.1.21451.13\Sdk\WasmApp.Native.targets(481,5): error : Precompiling failed for C:\Users\litvinov\source\repos\BlazorApp1\BlazorApp1\obj\Release\net6.0\linked\***AssemblyName***.dll [C:\Users\litvinov\source\repos\BlazorApp1\BlazorApp1\BlazorApp1.csproj]
```

We have hundreds of regexes in our library, so it is nearly impossible to determine which causes the problem. I tried the MONO_LOG_LEVEL=debug however without success. Do you have any ideas on how to find a workaround for this issue?

### Configuration
```
 Version:   6.0.100-rc.1.21458.32
 Commit:    d7c22323c4

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-rc.1.21458.32\

Host (useful for support):
  Version: 6.0.0-rc.1.21451.13
  Commit:  d7619cd4b1
```

You can find the whole log in the attachment.
[log.zip](https://github.com/dotnet/runtime/files/7249772/log.zip)

**Comment by vargaz** [comment]
Putting
    <ItemGroup>
      <MonoAOTCompilerDefaultProcessArguments Include="-v"/>
    <ItemGroup>

into your csproj will hopefully make the aot compiler print out all the methods before it compiles them so if you run a verbose build, it might be possible to find the method it fails on.

**Comment by SamMonoRT** [comment]
This is a dupe of old issue we reopened - https://github.com/dotnet/runtime/issues/57560 
We are working towards backporting a fix to 6.0
